### PR TITLE
Update pattern matching to fix a couple small issues

### LINKED ISSUE #1479: Pattern Matching: Check the vocabulary for "switch pattern expressions".  The 'switch expression' term may be wrong in the What's new article
The correct terms should be "Switch Statement" and "Match Expression" for the new syntax on the case statements.

### LINKED ISSUE #1585: Wrong information about pattern matching switch statement
In [https://docs.microsoft.com/en-us/dotnet/articles/csharp/csharp-7#pattern-matching], the documentation states that:

> The code above adds cases for `0` as a special case of `int`, and `null` as a special case when there is no input. This demonstrates one important new feature in switch pattern expressions: the order of the `case` expressions now matters. The `0` case must appear before the general `int` case. Otherwise, the first pattern to match would be the `int` case, even when the value is `0`.

Switching the `case 0` with `case int val` will be a `CS8120 The switch case has already been handled by a previous case.`

What works is adding the guard `when val != 0` to `case int val`.

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #1479
Fixes #1585

1. Update feature name for pattern matching (issue #1479)
2. Update description of match expressions to explain that match
expressions in the wrong order generate a compiler error (Issue #1585)
3. Make the C# 7 samples build and run under Core

### CODE REVIEW

File: samples/snippets/csharp/new-in-7/new-in-7.csproj [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp1.0</TargetFramework>
```
Comment (mairaw) [inline samples/snippets/csharp/new-in-7/new-in-7.csproj]: should we start using 1.1 based on the discussion we were having yesterday?

Comment (BillWagner) [inline samples/snippets/csharp/new-in-7/new-in-7.csproj]: Updated. Good catch.

**Review by rpetrusha — APPROVED** [review]
LGTM.

**Review by mairaw — APPROVED** [review]
:shipit:

**Pull request merged**